app/src/main/java/com/github/fribourgsdp/radio/MainActivity.kt

```diff
@@ -27,5 +33,28 @@ class MainActivity : AppCompatActivity() {
             startActivity(Intent(this, UserProfileActivity::class.java))
         }
 
+        /** this user allows quick demo's as it is data that is written to the app
+         * specific storage and can be easily read without intents */
+        try {
+            //if this is successful, we don't need to do anything
+            val user = User.load(this)
+
+        } catch (e: java.io.FileNotFoundException) {
+            createUser()
+        }
+
+
+    }
+    private fun createUser(){
```

Comment:
Are you sure it makes sense to create a user when you haven't logged in yet? The sign-in to firebase is useless with this.

Reply:
Yes, this way, a user doesn't need to connect in order to play. When a user comes in, he gets a generated user that allows him to join a game. The sign-in to firebase is not useless, as it allows a user to connect to multiple phones and to store its data in the cloud